### PR TITLE
Rework the documentation of our individual MAC implementations

### LINKED ISSUE #9709: MAC documenation needs updating
The individual MAC documentation files in `doc/man7` (i.e. `EVP_MAC_GMAC.pod` et al) are still described in terms of control functions not params.

### PR DESCRIPTION
**Pull request by levitte**
We now describe parameters instead of controls.

Also, since macros like EVP_MAC_CMAC do not exist any more, we rename
the pod files from EVP_MAC_{algo}.pod to EVP_MAC-{algo}.pod.  This
allows getting the documentation like this:

    man EVP_MAC CMAC

[skip ci]
Fixes #9709

### CODE REVIEW

File: doc/man3/EVP_MAC.pod [hunk 1]
```diff
@@ -167,49 +167,56 @@ and the given I<arg> as argument.
 
 =head1 PARAMETER NAMES
 
+Parameters are identified by name in form of strings, and have an
```
Comment (slontis) [inline doc/man3/EVP_MAC.pod]: NIT: in 'the' form

**Review by slontis — COMMENTED** [review]
Apart from a little NIT, it looks ok to me..
I assume you run make doc-nits if you are skipping testing?

**Comment by levitte** [comment]
> I assume you run make doc-nits if you are skipping testing?

Actually, that was to avoid CIs to do unnecessary work.  They *are* a finite resource, after all

**Comment by levitte** [comment]
However, a quick doc-nits revealed no errors

File: doc/man3/EVP_MAC.pod (hunk 1 above)
Comment (levitte) [reply doc/man3/EVP_MAC.pod]: Fixed

**Comment by levitte** [comment]
> "in the form of" --> "as"

Thanks.  Changed.

**Review by slontis — APPROVED** [review]

**Comment by levitte** [comment]
Ping?

**Comment by levitte** [comment]
Merged.

e592dbde6ff177a43cf66a95c59d101afb74cf5d Rework the documentation of our individual MAC implementations

**Pull request closed without merging**